File: scripts/toggle-floating.ts
#!/usr/bin/env bun

import { $ } from 'bun';
import { exit } from 'node:process';

const activeWindow = await $`hyprctl activewindow -j`.json();
const isFloating = activeWindow?.floating;

if (isFloating) {
  await $`hyprctl eval 'hl.dispatch(hl.dsp.window.float())'`;
  exit(0);
}

const windowW = 1600;
const windowH = 900;

await $`hyprctl eval 'hl.dispatch(hl.dsp.window.float())'`;

await $`hyprctl eval 'hl.dispatch(hl.dsp.window.resize({
    x = ${windowW},
    y = ${windowH}
}))'`;
await $`hyprctl eval 'hl.dispatch(hl.dsp.window.center())'`;

File: scripts/toggle-floating.lua
#!/usr/bin/env lua

local script_dir = arg[0]:match("(.*/)") or "./"
package.path = script_dir .. "?.lua;" .. script_dir .. "?/init.lua;" .. package.path

local hyprctl = require("lib.hyprctl")

local FLOAT_WIDTH = 1600
local FLOAT_HEIGHT = 900

if hyprctl.is_floating() then
    hyprctl.eval("hl.dsp.window.float()")
    os.exit(0)
end

hyprctl.eval("hl.dsp.window.float()")
hyprctl.eval(string.format(
    "hl.dsp.window.resize({ x = %d, y = %d })",
    FLOAT_WIDTH, FLOAT_HEIGHT
))
hyprctl.eval("hl.dsp.window.center()")
